lib/rubocop/cop/lint/shared_mutable_default.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for Hash creation with a mutable default value.
+      # Creating a Hash in such a way will share the default value
+      # across all keys, causing unexpected behavior when modifying it.
+      #
+      # For example, when the Hash was created with an Array as the argument,
+      # calling `hash[:foo] << 'bar'` will also change the value of all
+      # other keys that have not been explicitly assigned to.
+      #
+      # @example
+      #   # bad
+      #   Hash.new([])
+      #   Hash.new({})
+      #   Hash.new(Array.new)
+      #   Hash.new(Hash.new)
+      #
+      #   # okay -- In rare cases that intentionally have this behavior,
+      #     without disabling the cop, you can set the default explicitly.
```

Comment:
You need to make the second line a comment or else `rake documentation_syntax_check` tries to parse it 😄 
 
```suggestion
      #   # okay -- In rare cases that intentionally have this behavior,
      #   #   without disabling the cop, you can set the default explicitly.
```